src/toolkits/activity_classification/activity_classifier.cpp

```diff
@@ -652,15 +659,23 @@ std::unique_ptr<model_spec> activity_classifier::init_model() const
   size_t prediction_window = read_state<flex_int>("prediction_window");
   const flex_list &features_list = read_state<flex_list>("features");
 
-  // Initialize a random number generator for weight initialization.
-  std::seed_seq seed_seq = { read_state<int>("random_seed") };
-  std::mt19937 random_engine(seed_seq);
-
+  //Only to create a random engine for weight initialization if use_zero_init = false
+  std::mt19937 random_engine;
+  if (use_zero_init == false){
+    std::seed_seq seed_seq { read_state<int>("random_seed") };
+    random_engine = std::mt19937(seed_seq);
+  }
   result->add_channel_concat(
       "features",
       std::vector<std::string>(features_list.begin(), features_list.end()));
   result->add_reshape("reshape", "features",
                       {{1, num_features, 1, prediction_window}});
+  weight_initializer initializer;
+  lstm_weight_initializers lstm_initializer;
+  initializer = (use_zero_init) ? (weight_initializer) zero_weight_initializer() :
+  (weight_initializer) xavier_weight_initializer(num_features * prediction_window,
+                                NUM_CONV_FILTERS * prediction_window,
+                                &random_engine);
```

Comment:
I think you can just initialize `initializer` to a `zero_weight_initializer`, then only overwrite it when use random initialization.

Then we can avoid the unnecessary use of the ternary ?: operator